deeprvat/deeprvat/config.py

```diff
@@ -24,6 +28,228 @@ def cli():
     pass
 
 
+@cli.command()
+@click.argument("config-file", type=click.Path(exists=True))
+@click.argument("output-dir", type=click.Path(), default=".")
+def create_main_config(
+    config_file: str,
+    output_dir: str,
+):
+    """
+    Generates the necessary deeprvat_config.yaml file for running all pipelines.
+    This function expects inputs as shown in the following config-file:
+        - DEEPRVAT_DIR/example/deeprvat_input_config.yaml
+
+    :param config_file: Path to directory of relevant config yaml file
+    :type config_file: str
+    :param output_dir: Path to directory where created deeprvat_config.yaml will be saved.
+    :type output_dir: str
+    :return: Joined configuration file saved to deeprvat_config.yaml.
+    """
+    # Base Config
+    with open(
+        f"{os.path.dirname(deeprvat_dir.__file__)}/base_configurations.yaml"
+    ) as f:
+        base_config = yaml.safe_load(f)
+
+    expected_input_keys = [
+        "phenotypes_for_association_testing",
+        "phenotypes_for_training",
+        "gt_filename",
+        "variant_filename",
+        "phenotype_filename",
+        "annotation_filename",
+        "gene_filename",
+        "rare_variant_annotations",
+        "covariates",
+        "association_testing_data_thresholds",
+        "training_data_thresholds",
+        "seed_gene_results",
+        "training",
+        "n_repeats",
+        "y_transformation",
+        "evaluation",
+        "cv_exp",
+        "cv_path",
+        "n_folds",
+    ]
+
+    full_config = base_config
+
+    with open(config_file) as f:
+        input_config = yaml.safe_load(f)
+
+    # CV setup parameters
+    if not input_config["cv_exp"]:
+        logger.info("Not CV setup...removing CV pipeline parameters from config")
+        to_remove = {"cv_path", "n_folds"}
+        expected_input_keys = [
+            item for item in expected_input_keys if item not in to_remove
+        ]
+        full_config["cv_exp"] = False
+    else:
+        full_config["cv_path"] = input_config["cv_path"]
+        full_config["n_folds"] = input_config["n_folds"]
+        full_config["cv_exp"] = True
+
+    no_pretrain = True
+    if "use_pretrained_models" in input_config:
+        if input_config["use_pretrained_models"]:
+            no_pretrain = False
+            logger.info("Pretrained Model setup specified.")
+            to_remove = {"training", "phenotypes_for_training", "seed_gene_results"}
+            expected_input_keys = [
+                item for item in expected_input_keys if item not in to_remove
+            ]
+            expected_input_keys.extend(["use_pretrained_models", "model"])
+
+            with open(f"{os.path.dirname(pretrained_dir.__file__)}/config.yaml") as f:
+                pretrained_config = yaml.safe_load(f)
+
+            for k in pretrained_config:
+                input_config[k] = deepcopy(pretrained_config[k])
+
+    if set(input_config.keys()) - set(expected_input_keys):
```

Comment:
Great to check for this! Are you also checking for missing keys somewhere? I didn't see it.